cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1236,6 +1234,12 @@ private Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> onReady(Reconcilia
                 }, exception -> updateStatus(reconciliation, kafkaRebalance, new KafkaRebalanceStatus(), exception).mapEmpty());
     }
 
+    private boolean isKafkaClusterReady(Kafka kafka) {
```

Comment:
I think the code is actually correct, but the method name seems wrong. It seems to:
* Return true when the cluster is not ready
* Return false when it is ready

So you would either need to invert the condition inside this method or you would need to rename it to `isKafkaClusterNotReady` (the first is probably cleaner code with better readability).